compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -209,7 +209,7 @@ impl<'a> ArchiveBuilder<'a> for LlvmArchiveBuilder<'a> {
         let import_name_vector: Vec<CString> = dll_imports
             .iter()
             .map(if self.config.sess.target.arch == "x86" {
-                |import: &DllImport| CString::new(format!("_{}", import.name.to_string())).unwrap()
+                LlvmArchiveBuilder::i686_decorated_name
             } else {
                 |import: &DllImport| CString::new(import.name.to_string()).unwrap()
```

Comment:
Nit: it's a bit weird to return two closures from if/else, it's probably better to use one closure and process the `arch == "x86"` condition inside it.

Reply:
I meant `let import_name_vector: Vec<CString> = dll_imports.iter().map(new_single_function_for_both_64_bit_and_32_bit targets).collect();`

Reply:
Changed; I was just lifting the architecture test out of the loop since it's invariant, but I can't imagine that perf matters that much here.